astoria-tech/meetup
# ATM-39: Add Base Theme to Meetup Repo

# Description

- This PR adds a base theme to the repository.  It currently only has font families and colors. 

## Issue
[#39](https://github.com/astoria-tech/meetup/issues/39)

## Notes
- This PR does not include a dark theme as currently it is not in use.  However, it would be nice to have a dark theme at some point, this could be ticketed separately.
- The "gradients" mentioned in the issue were meaning like a gradient background on an element, but there are different tints of the `primary`, `secondary`, and `tertiary` colors named `primaryGradient`, `secondaryGradient`, and `tertiaryGradient`.  These can be renamed for clarity if desired.
- This PR uses locally hosted font files, could be somewhere on a CDN.  This could be a fast follow ticket if we prefer to go that direction.  (Also unfortunately did not use system fonts against my own suggestion.)
- The colors and font families were arbitrarily chosen from one of the existing flyers.  If we'd like a different set of colors or fonts, happy to switch things up!
- Unsure if the `postCSS` package is needed for imports, but added it so that directives could be imported and used as needed, where needed.   Could the reference to the `postCss` package  be in the `tailwind.config.mjs` file under plugins?
- Added `autoprefixer` for vendor prefixes.  If it feels unnecessary, for now I can remove it.
- Organized files for directives and theming into a new `src/styles` directory.
- Only minimally applied styles to elements in the feed, as that would be out of scope at the moment.

## How to test
- Pull this branch and fire it up with `npm run dev`
- See that the base font family is now `NotoSans`
- Add a color to any element from the themes, see it apply appropriately.

## Screenshots
Swatches:
<img width="481" alt="Screen Shot 2024-04-24 at 1 53 03 AM" src="https://github.com/astoria-tech/meetup/assets/3402228/dd2c4913-2898-4383-b701-16758a5afc0d">
Font Families:
<img width="188" alt="Screen Shot 2024-04-25 at 12 25 48 AM" src="https://github.com/astoria-tech/meetup/assets/3402228/92207f9a-8815-463b-91df-62f57c90a180">
Example of type elements applied on feed:
<img width="459" alt="Screen Shot 2024-04-25 at 12 57 14 AM" src="https://github.com/astoria-tech/meetup/assets/3402228/70c7f5e3-49a4-4640-b618-573857ff5c67">
